src/js/stores/__tests__/MesosStateStore-test.js

```diff
@@ -33,10 +33,102 @@ describe("MesosStateStore", function() {
       expect(result).toEqual([1, 2, 3]);
     });
 
-    it("nulls if no service matches", function() {
+    it("returns empty array if no service matches", function() {
       var result = MesosStateStore.getTasksFromServiceName("nonExistent");
       expect(result).toEqual([]);
     });
+
+    it("returns empty array for an invalid state", function() {
+      MesosStateStore.get = () => null;
+      var result = MesosStateStore.getTasksFromServiceName("marathon");
+      expect(result).toEqual([]);
+    });
+  });
+
+  describe("#getLastMesosState", function() {
+    beforeEach(function() {
+      this.get = MesosStateStore.get;
+    });
+
+    afterEach(function() {
+      MesosStateStore.get = this.get;
+    });
+
+    describe("fills empty fields with empty arrays", function() {
+      const testCases = [
+        [],
+        ["tasks"],
+        ["frameworks", "slaves"],
+        ["frameworks", "tasks", "executors"]
+      ];
+      const allFields = ["tasks", "frameworks", "slaves", "executors"];
+      console.log("ABC");
```

Comment:
💣

Reply:
🙈  Holy shit, I should read more carefully through my code before commiting it 😞